server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java

```diff
@@ -323,6 +341,60 @@ public T unremove(String name) {
         return unremovedChild;
     }
 
+    @Override
+    public void markForPurge(String name) {
+        ensureOpen();
+        validateChildName(name);
+        File marked;
+        final File removed = new File(rootDir, name + SUFFIX_REMOVED);
+
+        final Supplier<File> newMarkedFile = () -> {
+            final String interfix = '.' + Long.toHexString(ThreadLocalRandom.current().nextLong());
+            return new File(rootDir, name + interfix + SUFFIX_PURGED);
+        };
+
+        synchronized (this) {
+            if (!removed.exists() || !removed.isDirectory()) {
+                throw newStorageNotFoundException(name + SUFFIX_REMOVED);
+            }
+
+            marked = newMarkedFile.get();
+            while (!removed.renameTo(marked)) {
```

Comment:
I think it's possible to stuck here infinitely. How about using [`Files.move()`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#move-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-) and retrying only on `FileAlreadyExistsException`?

Reply:
Thanks, it is a nice idea.